File: solver_sample/agent.py
# 自チームは１、敵チームは２
# idは固定ではない
class Agent:
    def __init__(self, agent_id, x, y, team, done_butting, done_bad_act):
        self.x = x                          # int 0 ~ 盤面の幅 - 1
        self.y = y                          # int 0 ~ 盤面の高さ - 1
        self.agent_id = agent_id            # int 1試合の中で共通
        self.team = team                    # int 1(味方) or 2(敵)
        self.done_butting = done_butting    # bool 前のターンにバッティングを起こしたか
        self.done_bad_act = done_bad_act    # bool 無効な手を打ったか

File: 本番/Procon30/Agent.cpp
#include "Agent.h"

Agent::Agent()
{
	pos = Vector2(0, 0);
	delta_pos = Vector2(0, 0);
	act_type = 0;
	team = 0;
	done_butting = false;
	done_bad_act = false;
}

Agent::Agent(Vector2 _pos, int _team, int _ID)
{
	pos = _pos;
	delta_pos = Vector2(0, 0);
	act_type = 0;
	if (_team != 1 && _team != 2) {
		throw std::invalid_argument("Agent constructor exception.");
	}
	team = _team;
	ID = _ID;
	done_butting = false;
	done_bad_act = false;
}

void Agent::resetAct() {
	delta_pos = Vector2(0, 0);
	act_type = 0;
}
